appengine/pubsub/spec/pubsub_spec.rb

```diff
@@ -47,6 +50,41 @@ def app
     expect(last_response.status).to eq 303
   end
 
+  it "accepts a push" do
+    post "/pubsub/push?token=#{ENV["PUBSUB_VERIFICATION_TOKEN"]}",
+         JSON.generate({"message" => { "data" => Base64.encode64("A Message") }})
+
+    expect(last_response.status).to eq 200
+  end
+
+  it "accepts an authenticated push" do
+    public_cert_str = File.read "spec/fixtures/public_cert.pem"
+    key = OpenSSL::X509::Certificate.new(public_cert_str).public_key
+    key_info = Google::Auth::IDTokens::KeyInfo.new id: "test-key", key: key, algorithm: "RS256"
+    key_source = Google::Auth::IDTokens::StaticKeySource.new key_info
+    Google::Auth::IDTokens.instance_variable_set(:@oidc_key_source, key_source)
+
+    now = Time.now.to_i
+    jwt_payload = {
+      aud: 'example.com',
+      azp: '1234567890',
+      email: 'pubsub@example.iam.gserviceaccount.com',
+      email_verified: true,
+      iat: now,
```

Comment:
I'd suggest using something like `now - 60`, to prevent any flakiness that might come from clock skew.

Reply:
Done, thanks!